services/consumer/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"os"
+	"time"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	"github.com/red-hat-storage/ocs-operator/controllers/storagecluster"
+	providerclient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func main() {
+	flag.Parse()
+
+	scheme := runtime.NewScheme()
+	if err := ocsv1.AddToScheme(scheme); err != nil {
+		klog.Exitf("failed to add ocsv1 to scheme: %v", err)
+	}
+
+	config, err := config.GetConfig()
+	if err != nil {
+		klog.Exitf("failed to get config: %v", err)
+	}
+	client, err := client.New(config, client.Options{Scheme: scheme})
+	if err != nil {
+		klog.Exitf("failed to create controller-runtime client: %v", err)
+	}
+	name, isSet := os.LookupEnv(storagecluster.StorageClusterNameEnvVar)
+	if !isSet {
+		klog.Exit("STORAGE_CLUSTER_NAME env var not set")
+	}
+	namespace, isSet := os.LookupEnv(storagecluster.StorageClusterNamespaceEnvVar)
+	if !isSet {
+		klog.Exit("STORAGE_CLUSTER_NAMESPACE env var not set")
+	}
+
+	storageCluster := &ocsv1.StorageCluster{}
+	if err := client.Get(context.Background(), types.NamespacedName{Name: name, Namespace: namespace}, storageCluster); err != nil {
+		klog.Exit("failed to get storageCluster: %v", err)
+	}
+
+	providerClient, err := providerclient.NewProviderClient(
+		context.Background(),
+		storageCluster.Spec.ExternalStorage.StorageProviderEndpoint,
+		10*time.Second,
+	)
```

Comment:
I think you are missing a defer to make sure the client is closed if we exit on errors.